2996_reflection/reflection.md

```diff
@@ -6743,20 +6671,32 @@ consteval info type_of(info r);
 
 [#]{.pnum} *Constant When*: `$has-type$(r)` is `true`.
 
-[#]{.pnum} *Returns*: If `r` represents an value, object, or entity, then a reflection of the type of what is represented by `r`. Otherwise, if `r` represents a direct base class relationship, then a reflection of the type of the direct base class. Otherwise, for a data member description (`$T$`, `$N$`, `$A$`, `$W$`, `$NUA$`) ([class.mem.general]{.sref}), a reflection of the type `$T$`.
+[#]{.pnum} *Returns*:
+
+- [#.#]{.pnum} If `r` represents a value, object, variable, function, non-static data member, or bit-field, then the type of what is represented by `r`.
+- [#.#]{.pnum} Otherwise, if `r` represents an enumerator of an enumeration `$E$`, then:
+  - [#.#.#]{.pnum} If `$E$` is defined by a declaration `$D$` that is reachable from a point `$P$` in the evaluation context and `$P$` does not occur within an `$enum-specifier$` of `$D$`, then a reflection of `$E$`.
+  - [#.#.#]{.pnum} Otherwise, a reflection of the type of the enumerator prior to the closing brace of the `$enum-specifier$` as specified by [dcl.enum].
+- [#.#]{.pnum} Otherwise, if `r` represents a direct base class relationship, then a reflection of the type of the direct base class.
+- [#.#]{.pnum} Otherwise, for a data member description (`$T$`, `$N$`, `$A$`, `$W$`, `$NUA$`) ([class.mem.general]), a reflection of the type `$T$`.
 
 ```cpp
 consteval info object_of(info r);
 ```
 
-[#]{.pnum} *Constant When*: `r` represents either an object with static storage duration ([basic.stc.general]), or a variable associated with, or referring to, such an object.
+[#]{.pnum} *Constant When*: `r` is a reflection representing either
+
+- [#.#]{.pnum} an object with static storage duration ([basic.stc.general]), or
+- [#.#]{.pnum} a variable that either declares or refers to such an object, and if that variable is a reference `$R$` then either
+  - [#.#.#]{.pnum} `$R$` is usable in constant expressions ([expr.const]), or
+  - [#.#.#]{.pnum} the lifetime of `$R$` began during the core constant expression currently under evaluation.
 
-[#]{.pnum} *Returns*: If `r` represents a variable, then a reflection of the object associated with, or referred to by, the variable. Otherwise, `r`.
+[#]{.pnum} *Returns*: If `r` represents a variable, then a reflection of the object declared by, or referred to by, that variable. Otherwise, `r`.
 
 ::: example
 ```cpp
 int x;
-int& y = x;
+constexpr int& y = x;
```

Comment:
I thought you said this `constexpr` was unnecessary?

Reply:
Oops, yep. Correct.